scim/resource_service_principal.go

```diff
@@ -224,13 +224,13 @@ func ResourceServicePrincipal() common.Resource {
 				if isForceDeleteRepos {
 					err = workspace.NewNotebooksAPI(ctx, c).Delete(fmt.Sprintf("/Repos/%v", appId), true)
 					if err != nil {
-						return fmt.Errorf("force_delete_repos: %w", err)
+						return fmt.Errorf("force_delete_repos: %s", err.Error())
```

Comment:
This is needed due to changes in the behavior of `apierr.IsMissing()`. These errors were previously not considered to be missing because the status code used in the test is 400. However, in 0.39, this function changed to check whether the error inherits from `NotFound`. Because the status code returned from the API is RESOURCE_DOES_NOT_EXIST, this is considered missing. Using the error string here causes the error to not be considered missing.